lib/cfa/augeas_parser.rb

```diff
@@ -3,6 +3,23 @@
 require "cfa/placer"
 
 module CFA
+  # A building block for {AugeasTree}.
+  #
+  # Intuitively the tree is made of hashes where keys may be duplicated,
+  # so it is implemented as a sequence of hashes with two keys, :key and :value.
+  #
+  # A `:key` is a String.
+  # The key may have a collection suffix "[]". Note that in contrast
+  # with the underlying {::Augeas} library, an integer index is not present.
```

Comment:
Thanks, explaining reasons is a good thing. I'll just wait a bit until we clarify AugeasCollection as a whole.